### ISSUE TITLE
loss='categorical_crossentropy' VS loss=K.categorical_crossentropy

### ISSUE THREAD

**Issue by rjpg** [keras-team/keras#12155]
Hello!

Why I have very different loss values in training using these two lines code to define the loss function?   

```
modellstm.compile(loss=K.categorical_crossentropy, optimizer=OPTIMIZER,metrics=["accuracy"])
Output : 
59s - loss: 12.8045 - acc: 0.2530 - val_loss: 12.7778 - val_acc: 0.2289
```

and with :

```
modellstm.compile(loss='categorical_crossentropy', optimizer=OPTIMIZER,metrics=["accuracy"])
Output:
59s - loss: 1.6156 - acc: 0.2042 - val_loss: 1.6016 - val_acc: 0.2241
```

everything else is the same .... using loss=K.categorical_crossentropy the values (loss) are so high! 

What I am missing?   

Thanks in advance

**Comment by gabrieldemarmiesse** [comment]
Please provide a single script to put in evidence the bug, I can't reproduce the behavior you are mentioning. Thanks!

**Comment by rjpg** [comment]
Now I am testing in another computer with CUDA and they are similar (OK).
yesterday on the laptop, using cpu, they were different  ... I was running in jupyter notebook and maybe some keras/tf session was open before confusing things...

Anyway I will test again the script when I am with the laptop and post it here the results (if the bug persist I will upload my script and data to my github and give you the link)

**Comment by rjpg** [comment]
hello,

I have tested again. Using cpu the "bug" exist using cuda is ok

here is the notebook :

https://github.com/rjpg/bftensor/blob/master/Autoencoder/src/testing%20loss.ipynb

`using modellstm2.compile(loss=K.categorical_crossentropy, optimizer=OPTIMIZER,metrics=["accuracy"])`

The loss is too high :

```
Epoch 1/5
53s - loss: 12.8783 - acc: 0.2114 - val_loss: 12.8372 - val_acc: 0.2361
Epoch 2/5
59s - loss: 12.7791 - acc: 0.2446 - val_loss: 12.7565 - val_acc: 0.2193
Epoch 3/5
62s - loss: 12.6016 - acc: 0.2494 - val_loss: 12.6343 - val_acc: 0.2217
Epoch 4/5
57s - loss: 12.3212 - acc: 0.2452 - val_loss: 12.3410 - val_acc: 0.2410
Epoch 5/5
64s - loss: 12.0311 - acc: 0.2693 - val_loss: 12.2590 - val_acc: 0.2337
```

Using CUDA they are similar OK

Thanks

**Comment by tranvohuy** [comment]
Just a report for future investigators. I've run rjpg's code and could not see the bug. My code ran in google colab, tested with cpu and gpu. I took random inputs and outputs
```
total_inputs = np.random.randn(1000,128,9)
total_output = np.random.randint(5, size=(1000,1))
```

I've looked at the codes [training.py](https://github.com/keras-team/keras/blob/master/keras/engine/training.py#L142), [losses.py](https://github.com/keras-team/keras/blob/master/keras/losses.py#L131), and [generic_utils.py](https://github.com/keras-team/keras/blob/master/keras/utils/generic_utils.py#L120) and found nothing suspicious.
